samcli/cli/types.py

```diff
@@ -9,15 +9,47 @@
 import click
 
 
-def _value_regex(delim):
-    return f'(\\"(?:\\\\.|[^\\"\\\\]+)*\\"|(?:\\\\.|[^{delim}\\"\\\\]+)+)'
+def _match_regex(match, delim):
```

Comment:
Can you throw a quick doc string on this? Finding it slightly hard to understand what/why you would need the match here.

Reply:
Will do